src/comp.rs

```diff
@@ -197,7 +173,21 @@ pub fn run_compositor(
 			.start_process(
 				Process::new()
 					.with_executable("cosmic-comp")
-					.with_env([("COSMIC_SESSION_SOCK", comp.as_raw_fd().to_string())]),
+					.with_env([("COSMIC_SESSION_SOCK", comp.as_raw_fd().to_string())])
+					.with_on_exit(move |pman, _, err_code, _will_restart| {
+						let session_dbus_tx = session_dbus_tx.clone();
+						async move {
+							pman.stop();
+							if err_code == Some(0) {
+								info!("cosmic-comp exited successfully");
```

Comment:
I don't think we should restart with the compositor exited normally. This would break the default `Terminate`-shortcut, no? If anything we should send an `Exit` Request, because right now this case doesn't always cleanly return to the login manager.